File: aat-android/src/main/java/ch/bailu/aat/gpx/segmented_list/SegmentNode.java
package ch.bailu.aat.gpx.segmented_list;

import ch.bailu.aat.gpx.linked_list.Node;

public class SegmentNode extends Node {
    private final SegmentNode marker;

    private final Node first;
    private int count=0;



    public SegmentNode(Node n, SegmentNode m) {
        first=n;
        marker=m;
    }


    public SegmentNode(Node n) {
        this(n,null);
    }


    public Node getFirstNode() {
        return first;
    }


    public void update(Node n) {
        count++;
    }


    public int getSegmentSize() {
        return count;
    }


    public SegmentNode getMarker() {
        return marker;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/segmented_list/SegmentNode.kt
package ch.bailu.aat_lib.gpx.segmented_list

import ch.bailu.aat_lib.gpx.linked_list.Node

open class SegmentNode(val firstNode: Node, val marker: SegmentNode? = null) : Node() {
    var segmentSize = 0
        private set

    open fun update(node: Node) {
        segmentSize++
    }
}
